book/listings/todo_app/1/window/mod.rs

```diff
@@ -1,32 +1,32 @@
 mod imp;
 
-use std::fs::File;
-
-use crate::todo_object::{TodoData, TodoObject};
+use crate::todo_object::TodoObject;
 use crate::todo_row::TodoRow;
-use crate::utils::data_path;
 use glib::{clone, Object};
 use gtk::prelude::*;
 use gtk::subclass::prelude::*;
 use gtk::{gio, glib};
-use gtk::{Application, CustomFilter, FilterListModel, NoSelection, SignalListItemFactory};
-use log::info;
+use gtk::{Application, NoSelection, SignalListItemFactory};
 
+// ANCHOR: glib_wrapper
 glib::wrapper! {
     pub struct Window(ObjectSubclass<imp::Window>)
-        @extends gtk::Widget, gtk::Window, gtk::ApplicationWindow,
-        @implements gio::ActionMap, gio::ActionGroup;
+    @extends gtk::ApplicationWindow, gtk::Window, gtk::Widget,
+    @implements gio::ActionGroup, gio::ActionMap, gtk::Accessible, gtk::Buildable,
+                gtk::ConstraintTarget, gtk::Native, gtk::Root, gtk::ShortcutManager;
 }
+// ANCHOR_END: glib_wrapper
 
 impl Window {
     pub fn new(app: &Application) -> Self {
         // Create new window
         Object::new(&[("application", app)]).expect("Failed to create `Window`.")
     }
 
+    // ANCHOR: model
     fn model(&self) -> &gio::ListStore {
         // Get state
-        let imp = imp::Window::from_instance(&self);
+        let imp = imp::Window::from_instance(self);
         imp.model.get().expect("Could not get model")
```

Comment:
I'm not sure what the point of expect messages like this is. What does it add over `unwrap` other than boilerplate?

Reply:
> What does it add over unwrap other than boilerplate?

In this specific case indeed not much.
I have the hope that newcomers get more out of the concept of `expect` than from unwraps alone.
Then I gave myself the freedom to use `expect` everywhere instead of deciding on a per-case basis